src/convertToObject.js

```diff
@@ -6,7 +6,69 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (!sourceString.length) {
+    return {};
+  }
+
+  const rawLines = sourceString.split(';');
+
+  // filter lines without letters
+  const noEmptyLines = rawLines
+    .map((line) => {
+      return line.trim();
+    })
+    .filter((item) => item.length > 0);
+
+  const aggregatedLines = [];
+
+  for (const line of noEmptyLines) {
+    if (line.includes(':')) {
+      aggregatedLines.push(line);
+    } else {
+      aggregatedLines[aggregatedLines.length - 1] += line;
+    }
+  }
+
+  let separatedRules = [];
+
+  for (const line of aggregatedLines) {
+    if (line.split(':').length >= 3) {
+      separatedRules = [...separatedRules, ...line.split(';')];
+    } else {
+      separatedRules.push(line);
+    }
+  }
+
+  separatedRules = separatedRules.filter((item) => item.length > 1);
+
+  const convertStrToObj = separatedRules.map((item) => {
+    const [property, value] = item.split(':');
+
+    return { [property]: value };
+  });
+
+  // remove ';'
+  convertStrToObj.forEach((item, index, array) => {
+    const key = Object.keys(item)[0];
+    const newValue = item[key].split(';').join('');
+
+    array[index] = { [key]: newValue };
```

Comment:
[CODE KNOWLEDGE] - Don't use iteration methods to modify an already existing array/object. The `forEach` method here is incorrectly used to mutate the array. Instead, you should use `reduce` or `map` to return a new array.